game/cards/dm07/spells.go

```diff
@@ -0,0 +1,339 @@
+package dm07
+
+import (
+	"duel-masters/game/civ"
+	"duel-masters/game/cnd"
+	"duel-masters/game/fx"
+	"duel-masters/game/match"
+	"fmt"
+)
+
+func HopelessVortex(c *match.Card) {
+	c.Name = "Hopeless Vortex"
+	c.Civ = civ.Darkness
+	c.ManaCost = 5
+	c.ManaRequirement = []string{civ.Darkness}
+
+	c.Use(fx.Spell, fx.When(fx.SpellCast, func(card *match.Card, ctx *match.Context) {
+
+		creatures := fx.Select(card.Player, ctx.Match, ctx.Match.Opponent(card.Player), match.BATTLEZONE, "Destroy one of your opponent's creatures", 1, 1, false)
+
+		for _, creature := range creatures {
+
+			ctx.Match.Destroy(creature, card, match.DestroyedBySpell)
+
+		}
+	}))
+}
+
+func FreezingIcehammer(c *match.Card) {
+
+	c.Name = "Freezing Icehammer"
+	c.Civ = civ.Nature
+	c.ManaCost = 3
+	c.ManaRequirement = []string{civ.Nature}
+
+	c.Use(fx.Spell, fx.When(fx.SpellCast, func(card *match.Card, ctx *match.Context) {
+
+		fx.SelectFilter(
+			card.Player,
+			ctx.Match,
+			ctx.Match.Opponent(card.Player),
+			match.BATTLEZONE,
+			"Select 1 of your opponent's water or darkness creatures and put it in their manazone",
+			1,
+			1,
+			false,
+			func(x *match.Card) bool { return x.Civ == civ.Water || x.Civ == civ.Darkness },
+		).Map(func(x *match.Card) {
+			x.Player.MoveCard(x.ID, match.BATTLEZONE, match.MANAZONE)
+			ctx.Match.Chat("Server", fmt.Sprintf("%s was moved to %s's manazone", x.Name, x.Player.Username()))
```

Comment:
All the ".Chat" calls need to be replaced by ".ReportActionInChat", which also take a player as a param. The player passed should be the owner of the moved card, if movement happens.